### ISSUE TITLE
rpc: add metrics for client-side and server-side RPC request latencies and log events for latency anomalies

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#66777]
This is related to #65753. 
It came up while investigating #66550 / #66768 
and also would have helped @sumeerbhola investigate a customer issue

**tldr: we want metrics for RPC request latency, as well as log events for "slow" RPCs**

Detail:

- metrics:
  - RPC service latency, as measured from a RPC client, from point where a RPC request is sent and the point the RPC response is received. Takes into account both server-side latency and network roundtrip.
  - RPC server latency, as measured from ingress of RPC request to egress of the response, for every RPC handler. Measures the time needed to serve RPC request, and may include recursive RPC calls for fan-outs.
  - we probably need to define separate metric per RPC endpoint, as they have separate latency classes. (unsure)

- logging:
  - for any given RPC request, from the perspective of a client, if the request takes >50% more time  or <50% less time than the last call, log an event.

**Comment by tbg** [comment]
Is this logging, in its entirety, useful for RPCs in general? For example, the `Node.Batch` endpoint has wildly different latency expectations. A point read without contention is going to be fast, a point read with contention is going to be much slower.

**Comment by knz** [comment]
> A point read without contention is going to be fast, a point read with contention is going to be much slower.

1. that's why I was suggesting to have different logging/metrics per endpoint
2. we will probably welcome visibility into a metric that reacts to contention, so in your particular example it's a good thing

(A better example would be RPC calls which are inherently dependent on data size or some other thing that wouldn't provide a  useful signal. Again, separate metrics will help us learn what's useful and what is not.)
